presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusStandardizedRow.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import io.prestosql.spi.block.Block;
+
+public class PrometheusStandardizedRow
+{
+    Block labels;
+    Double timestamp;
```

Comment:
Why is this a double instead of a Timestamp?

Reply:
It seemed an okay way to represent the value coming back from Prometheus internally. A `long` might actually be the correct type based on a discussion of the TimestampType in Presto developer docs. But Double does have the width to handle the time value. Because we declare that column to be of type TimestampType, Presto eventually handles the conversion.

Reply:
In the end we're calling `(double) prometheusTimeSeriesValue.timestamp.toEpochMilli()` which just seems unecessary if we can pass the timestamp directly, so that's why im asking. Does presto not support the java timestamp type when using TimestampType.TIMESTAMP?

Reply:
It can be java.sql.Timestamp if we can figure out the changed usage of the Jackson deserialization required. Diving in now.

Reply:
Hmm, so I did all the conversion work, wrote a deserialize method, but in the end, I believe the right answer is actually: Presto wants timestamps to show up as Double so it can convert them itself:

```presto> SELECT * FROM prometheus.default.up ORDER BY timestamp ASC LIMIT 10;

Query 20190924_181523_00001_pmrz4, FAILED, 1 node
Splits: 38 total, 0 done (0.00%)
0:01 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20190924_181523_00001_pmrz4 failed: java.sql.Timestamp cannot be cast to java.lang.Double```

Reply:
@bentito Can you confirm with the Presto devs?

Reply:
yep. going to try to research a bit before I ask too.

Reply:
Actually I think I found the answer in AccumuloRecordCursor:
```        checkFieldType(field, BIGINT, DATE, INTEGER, REAL, SMALLINT, TIME, TIMESTAMP, TINYINT);
        Type type = getType(field);
        if (type.equals(BIGINT)) {
            return serializer.getLong(fieldToColumnName[field]);
        }
        else if (type.equals(DATE)) {
            return MILLISECONDS.toDays(serializer.getDate(fieldToColumnName[field]).getTime());
        }
        else if (type.equals(INTEGER)) {
            return serializer.getInt(fieldToColumnName[field]);
        }
        else if (type.equals(REAL)) {
            return Float.floatToIntBits(serializer.getFloat(fieldToColumnName[field]));
        }
        else if (type.equals(SMALLINT)) {
            return serializer.getShort(fieldToColumnName[field]);
        }
        else if (type.equals(TIME)) {
            return serializer.getTime(fieldToColumnName[field]).getTime();
        }
        else if (type.equals(TIMESTAMP)) {
            return serializer.getTimestamp(fieldToColumnName[field]).getTime();
        }
        else if (type.equals(TINYINT)) {
            return serializer.getByte(fieldToColumnName[field]);
        }
        else {
            throw new PrestoException(NOT_SUPPORTED, "Unsupported type " + getType(field));
        }``` So we can carry a timestamp as any type we want (and I'll swap back in the code that uses Timestamp throughout, cuz it _is_ more clear). We just need to make it a `long` and convert it based on the column type in `getLong`.

Reply:
Done in cdb254b